### PR TITLE
[Five 165] Mostrar confirmacion antes de borrar un artefacto

### PR DESCRIPTION
**Pull request by gperelstein**
Descripción
Se debe mostrar un cartel de confirmación antes de borrar un artefacto de un proyecto

Cambios realizados:

- Se añadió un componente para mostrar el cartel de confirmación antes de borrar un artefacto de un proyecto.

- Se instaló el paquete sweetalert2 para mostrar un cartel de exito o error al momento de borrar un artefacto

### CODE REVIEW

File: FiveRockingFingers/FRF.Web/ClientApp/src/components/ArtifactsComponents/ConfirmationDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                <DialogContent>
+                    <DialogContentText id="alert-dialog-description">
+                        El proyecto <i>"{props.artifactToDelete.name}</i>" será eliminado
+                    </DialogContentText>
+                </DialogContent>
+                <DialogActions>
+                    <Button onClick={handleConfirm} color="primary">
+                        Confirmar
+                    </Button>
+                    <Button onClick={handleCancel} color="primary" autoFocus>
```
Comment (matiasbeckerle) [inline FiveRockingFingers/FRF.Web/ClientApp/src/components/ArtifactsComponents/ConfirmationDialog.tsx:50]: Primary for the secondary action? I think you could make use of a secondary color instead.

**Review by matiasbeckerle — APPROVED** [review]

**Comment by matiasbeckerle** [comment]
Just remembered: don't you have a modal dialog utility in the bootstrap or material-ui libraries that you're already using folks? Have in mind that adding libraries is expensive.

**Comment by gperelstein** [comment]
I didn't know that adding libraries is expensive. I include it because I thought it could help to simplify the code and the modals dialogs look great. I think there is nothing like this in the libraries we already using but I can use a toast to send the message to the user.

**Comment by matiasbeckerle** [comment]
It's a tradeoff. Adding libraries is not something you should do slightly. Yes, on one hand it helps you to speed up the development of some functionality but...

- Then you and all team members need to know how to use it.
- Need to keep it updated for improvements or security vulnerabilities.
- What if you need to do something custom? You will have to hack around it and learn more deeply about it.
- What if they end support (something that happens quite often lately), are you going to fix the issues?
- How much the webapp footprint will increase in size? Some libraries are small... but each one of them counts to the overall page size. That's one of the reason some pages takes ages to load: you have to download big chunks of libraries.
- It's popular? It's maintained or there are tons of issues created and not addressed?
- Is is worthy to add 10k lines of code just to calculate the square root of a value or maybe I can do it with a custom 3 lines function by myself? (this last one is an exaggeration but there are a lot of examples and memes of js libraries doing basic stuff that you could easily resolve and avoid to add a big library)

Those are some of the questions you should ask before adding one.

TL;DR: I'm not telling you to not add it. I'm trying to make you evaluate it for yourself and ask those questions. And since bootstrap and material ui are quite big libraries, maybe you have this functionality available? I don't know, you tell me.

**Comment by gperelstein** [comment]
I understand, I didn't take into consideration all of these points. Thank you for the advice! I searched for something like the modals message that this library provides and I didn't find anything. But I can do something quite similar with material-ui. The code will be a little more complicated but taking the points that you remark into consideration I think it will pay off.

File: FiveRockingFingers/FRF.Web/ClientApp/src/components/ArtifactsComponents/ArtifactsTableRow.tsx [hunk 2]
```diff
@@ -1,22 +1,31 @@
...
 import Artifact from '../../interfaces/Artifact';
 import { Button } from 'reactstrap';
+import ConfirmationDialog from './ConfirmationDialog';
 
 const ArtifactsTableRow = (props: { artifact: Artifact, deleteArtifact: Function }) => {
 
+    const [openConfirmDialog, setOpenConfirmDialog] = React.useState(false);
+
+    
+
```
[the remark this answers is not in the document]
Comment (gperelstein) [reply FiveRockingFingers/FRF.Web/ClientApp/src/components/ArtifactsComponents/ArtifactsTableRow.tsx]: Done!

File: FiveRockingFingers/FRF.Web/ClientApp/src/components/ArtifactsComponents/ConfirmationDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
+﻿import { Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle, Snackbar } from '@material-ui/core';
+import { Alert } from 'reactstrap';
+import axios from 'axios';
+import * as React from 'react';
+import Artifact from '../../interfaces/Artifact';
+import Swal from 'sweetalert2'
+
+
```
[the remark this answers is not in the document]
Comment (gperelstein) [reply FiveRockingFingers/FRF.Web/ClientApp/src/components/ArtifactsComponents/ConfirmationDialog.tsx]: Done

File: FiveRockingFingers/FRF.Web/ClientApp/src/components/ArtifactsComponents/ConfirmationDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+const ConfirmationDialog = (props: { open: boolean, setOpen: Function, deleteArtifact: Function, artifactToDelete: Artifact }) => {
+
+
+
```
[the remark this answers is not in the document]
Comment (gperelstein) [reply FiveRockingFingers/FRF.Web/ClientApp/src/components/ArtifactsComponents/ConfirmationDialog.tsx:6]: Done!

File: FiveRockingFingers/FRF.Web/ClientApp/src/components/ArtifactsComponents/ConfirmationDialog.tsx (hunk 1 above)
Comment (gperelstein) [reply FiveRockingFingers/FRF.Web/ClientApp/src/components/ArtifactsComponents/ConfirmationDialog.tsx]: Done!

**Comment by gperelstein** [comment]
I already uninstall the library a had added. I use material-ui to send a Snackbar with the information to the user

File: FiveRockingFingers/FRF.Web/ClientApp/src/commons/SnackbarMessage.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
+﻿import * as React from 'react';
+import { Snackbar } from '@material-ui/core';
+import MuiAlert, { AlertProps } from '@material-ui/lab/Alert';
+
+function Alert(props: AlertProps) {
+    return <MuiAlert elevation={6} variant="filled" {...props} />;
+}
+
+const SnackbarMessage = (props: { message: string, severity: "success" | "info" | "warning" | "error" | undefined, open: boolean, setOpen: Function }) => {
```
Comment (matiasbeckerle) [inline FiveRockingFingers/FRF.Web/ClientApp/src/commons/SnackbarMessage.tsx:9]: Classname (and filename) is kind of hilarious, makes me thing on food. Should you use words like modal, dialog, popup, message, alertbox?

**Review by matiasbeckerle — APPROVED** [review]
I liked how easy you solved it. Good job.

File: FiveRockingFingers/FRF.Web/ClientApp/src/commons/SnackbarMessage.tsx (hunk 5 above)
Comment (gperelstein) [reply FiveRockingFingers/FRF.Web/ClientApp/src/commons/SnackbarMessage.tsx:9]: Hehehe you are right! The name is odd but I just use the naming that is used in material-ui library. I can change the name but I think that with this name everybody is going to know that I create this component using the Snackbar element from material-ui as the base, what do you think?

Comment (matiasbeckerle) [reply FiveRockingFingers/FRF.Web/ClientApp/src/commons/SnackbarMessage.tsx:9]: It's fair and that's actually a good argument. Leave it that way unless you (the ones working on the project) change your mind.
